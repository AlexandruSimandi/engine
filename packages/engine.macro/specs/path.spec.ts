import pluginTester from "babel-plugin-tester";
import prettier from "prettier";
import plugin from "babel-plugin-macros";

const macroFile = "'@c11/engine.macro'";
const config = require("./babelConfig.json");

pluginTester({
  plugin,
  babelOptions: { filename: __filename, ...config },
  formatResult: (result: any) => {
    return prettier.format(result, {
      parser: "babel",
    });
  },
  tests: {
    "should support Path": {
      code: `
        import { Path } from ${macroFile};
        Path.foo.bar
      `,
      snapshot: true,
    },
  },
});
