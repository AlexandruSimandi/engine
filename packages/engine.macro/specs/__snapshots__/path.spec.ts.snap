// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros should support Path: should support Path 1`] = `

import { Path } from '@c11/engine.macro';
Path.foo.bar

      ↓ ↓ ↓ ↓ ↓ ↓

Path.foo.bar;


`;
